src/main/scala/com/spotify/ratatool/samplers/BigSampler.scala

```diff
@@ -0,0 +1,346 @@
+        }
+      }
+    }
+  }
+  // scalastyle:on cyclomatic.complexity
+
+  private def sampleBigQueryTable(sc: ScioContext,
+                                  inputTbl: TableReference,
+                                  outputTbl: TableReference,
+                                  fields: List[String],
+                                  samplePct: Float,
+                                  seed: Option[Int]): Future[Tap[TableRow]] = {
+    val patchedBigQueryService = new PatchedBigQueryService(sc.optionsAs[BigQueryOptions])
+
+    if (patchedBigQueryService.getTable(outputTbl) != null) {
+      log.info(s"Reuse previous sample at $outputTbl")
+      Taps().bigQueryTable(outputTbl)
+    } else {
+      log.info(s"Will sample from BigQuery table: ${inputTbl}, output will be ${outputTbl}")
+
+      val schema = patchedBigQueryService.getTable(inputTbl).getSchema
+
+      // defeat closure TableSchema and TableFieldSchema are not serializable
+      val schemaStr = JsonSerDe.toJsonString(schema)
+      @transient lazy val schemaFields =
+        JsonSerDe.fromJsonString(schemaStr, classOf[TableSchema]).getFields.asScala.toList
+      val (s, n) = samplePctToInt(samplePct, 100)
+      val r = sc.bigQueryTable(inputTbl)
+        .flatMap { e =>
+          val hash = hashTableRow(
+            e,
+            schemaFields,
+            fields,
+            kissHashFun(seed).newHasher()).hash()
+          diceElement(e, hash, s.toInt, n)
+        }
+        .saveAsBigQuery(outputTbl, schema, WRITE_EMPTY, CREATE_IF_NEEDED)
+      sc.close()
+      r
+    }
+  }
+
+  private def sampleAvro(sc: ScioContext,
+                         input: String,
+                         output: String,
+                         sampleClass: Class[_],
+                         samplePct: Float,
+                         seed: Option[Int]): Future[Tap[GenericRecord]] = {
+    val schema = AvroIO.getAvroSchemaFromFile(input.toString)
+    val outputParts =
+      if (output.endsWith("/")) output + "part*" else output + "/part*"
+    if (FileStorage(outputParts).isDone) {
+      // sampling done previously - just reuse
+      log.info(s"Reuse previous sample at $outputParts")
+      Taps().avroFile(outputParts, schema = schema)
+    } else {
+      log.info(s"Will sample from: $input, output will be $output")
+      @transient lazy val sampleSchema = sampleClass.
+        asInstanceOf[Class[GenericRecord]].newInstance().getSchema
+      val (s, n) = samplePctToInt(samplePct, 100)
+      val r = sc.avroFile[GenericRecord](input, schema)
+        .flatMap { e =>
+          val hash =
+            hashAvroField(e, sampleSchema.getFields.asScala, kissHashFun(seed).newHasher()).hash()
+          diceElement(e, hash, s.toInt, n)
+        }
+        .saveAsAvroFile(output, schema = schema)
+      sc.close()
+      r
+    }
+  }
+
+  @tailrec
+  private def samplePctToInt(samplePct: Float, n: Int): (Float, Int) = {
```

Comment:
This will (in many cases) produce an approximate sample size, right? Is that the intention? For example `samplePctToInt(0.1999, 100)` would be `(1.999, 1000)` and later gets floored down to `(1, 1000)` right? I guess at most this will be off by one, so this is 👍